src/server/trace/viewer/traceViewer.ts

```diff
@@ -196,6 +196,23 @@ async function appendTraceEvents(model: TraceModel, file: string) {
 }
 
 export async function showTraceViewer(tracePath: string, browserName: string, headless = false): Promise<BrowserContext | undefined> {
+  const dir = fs.mkdtempSync(path.join(os.tmpdir(), `playwright-trace`));
+  process.on('exit', () => rimraf.sync(dir));
+
+  if (/^https?:\/\//i.test(tracePath)){
+    const downloadZipPath = path.join(dir, 'trace.zip');
+    try {
+      await download(tracePath, downloadZipPath, {
```

Comment:
I have played with this locally:
- Let's pass `progressBarName: tracePath`.
- Let's pass `log: debugLogger.log.bind(debugLogger, 'install')` - otherwise it is too chatty.
- Let's remove `Download failed.` prefix below - the error message already says "Download failed" 😄 

The rest is great, works like a charm!

Reply:
Thanks, done 😄